### PR TITLE
[MT7] Register more blockish tags MTC-27461

### PR DESCRIPTION
**Pull request by charsbar**

### CODE REVIEW

File: lib/MT/Util.pm [hunk 1]
```diff
@@ -700,9 +700,9 @@ sub _change_lf {
 sub html_text_transform {
     my $str = shift;
     $str = '' unless defined $str;
-    my $tags = qr!(?:h1|h2|h3|h4|h5|h6|table|ol|dl|ul|li|menu|dir|p|pre|center|form|fieldset|select|blockquote|address|div|hr|script|style)!;
+    my $tags = qr!(?:h1|h2|h3|h4|h5|h6|table|ol|dl|ul|li|menu|dir|p|pre|center|form|fieldset|select|blockquote|address|div|hr|script|style|article|aside|details|dialog|figcaption|figure|footer|header|hgroup|main|map|nav|section|template|thead|tfoot|tbody|tr|th|td|caption|colgroup|col|dt|dd|optgroup|legend|menuitem|option|param|summary|area|svg)!;
     $str =~ s/\r\n/\n/gs;
-    my $special_tags = qr!(?:script|style|pre)!;
+    my $special_tags = qr!(?:script|style|pre|svg)!;
```
Comment (usualoma) [inline lib/MT/Util.pm:705]: I think `map` is a special tag.

```suggestion
    my $special_tags = qr!(?:script|style|pre|map|svg)!;
```

File: lib/MT/Util.pm [hunk 2]
```diff
@@ -700,9 +700,9 @@ sub _change_lf {
 sub html_text_transform {
     my $str = shift;
     $str = '' unless defined $str;
-    my $tags = qr!(?:h1|h2|h3|h4|h5|h6|table|ol|dl|ul|li|menu|dir|p|pre|center|form|fieldset|select|blockquote|address|div|hr|script|style)!;
+    my $tags = qr!(?:h1|h2|h3|h4|h5|h6|table|ol|dl|ul|li|menu|dir|p|pre|center|form|fieldset|select|blockquote|address|div|hr|script|style|article|aside|details|dialog|figcaption|figure|footer|header|hgroup|main|map|nav|section|template|thead|tfoot|tbody|tr|th|td|caption|colgroup|col|dt|dd|optgroup|legend|menuitem|option|param|summary|area|svg)!;
```
Comment (usualoma) [inline lib/MT/Util.pm:703]: I think `svg` should not be included in the `$tag`. Because the browser renders `svg` as an inline element.

Comment (usualoma) [reply lib/MT/Util.pm:703]: I think
```html
a
<svg viewBox="0 0 10 10" xmlns="http://www.w3.org/2000/svg" stroke="red" fill="grey">
  <circle cx="5" cy="5" r="10" />
</svg>
b
```
should be
```html
a<br />
<svg viewBox="0 0 10 10" xmlns="http://www.w3.org/2000/svg" stroke="red" fill="grey">
  <circle cx="5" cy="5" r="10" />
</svg><br />
b<br />
```

File: lib/MT/Util.pm (hunk 1 above)
Comment (charsbar) [reply lib/MT/Util.pm]: thanks. added map and several others to $special_tags

File: lib/MT/Util.pm (hunk 2 above)
Comment (charsbar) [reply lib/MT/Util.pm]: thanks. I wonder if people really want to wrap multi-lined svg with p, but the point is taken.

**Pull request merged**